### PR TITLE
Add span overloads for Image.Load() and Image.DetectFormat()

### PR DESCRIPTION
**Pull request by antonfirsov**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
This PR implements API-s proposed in #565 by @jherby2k with the help of [System.IO.UnmanagedMemoryStream](https://www.nuget.org/packages/System.IO.UnmanagedMemoryStream/) which is a new dependency for targets > netstandard 1.1.

Related tests have been refactored heavily (isolation, cleanup, code share).

### CODE REVIEW

File: src/ImageSharp/ImageSharp.csproj [hunk 1]
```diff
@@ -44,6 +44,9 @@
     <PackageReference Include="System.Memory" Version="4.5.0" />
     <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="4.5.0" />
   </ItemGroup>
+  <ItemGroup Condition=" '$(TargetFramework)' != 'netstandard1.1' ">
```
Comment (tocsoft) [inline src/ImageSharp/ImageSharp.csproj]: you can skip this for our net standard 2.0 > targets too so you probably want to change this to `'$(TargetFramework)' == 'netstandard1.3'` instead

**Comment by antonfirsov** [comment]
@tocsoft good catch, fixed!

File: src/ImageSharp/Image.FromBytes.cs [hunk 2]
```diff
@@ -171,5 +172,125 @@ public static Image<TPixel> Load<TPixel>(Configuration config, byte[] data, IIma
...
+        /// By reading the header on the provided byte array this calculates the images format.
+        /// </summary>
+        /// <param name="config">The configuration.</param>
+        /// <param name="data">The byte array containing encoded image data to read the header from.</param>
+        /// <returns>The mime type or null if none found.</returns>
+        public static unsafe IImageFormat DetectFormat(Configuration config, ReadOnlySpan<byte> data)
+        {
+            fixed (byte* ptr = &data.GetPinnableReference())
+            {
+                using (var stream = new UnmanagedMemoryStream(ptr, data.Length))
```
Comment (davidfowl) [inline src/ImageSharp/Image.FromBytes.cs]: It's unfortunate to allocate a heap object to wrap the span...

Comment (antonfirsov) [reply src/ImageSharp/Image.FromBytes.cs]: Makes sense in the `DetectFormat()` use case, thanks!

When doing a full image decoding (`Image.Load(...)`), that allocation is a drop in the bucket however. Using `UnmanagedMemoryStream` seemed to be the best way for "converting" a span into a `Stream`.

**Pull request merged**